### PR TITLE
Consolidate tax configuration.

### PR DESCRIPTION
**Pull request by SteveIreland**
This consolidates the tax configuration options into a single configuration panel.

There is a minor database change to the tax_categories table.  I overlooked setting the primary key to auto_increment, so this pull request corrects that.

### CODE REVIEW

**Comment by SteveIreland** [comment]
Found a flaw when I started testing for failure instead of success (not my preferred testing mode).  So I added the fix to the pull request.

**Comment by SteveIreland** [comment]
Ack, apparently one of my "fixes" broke something else.  Need a little more time to figure out why.

File: application/controllers/Config.php [hunk 1]
```diff
@@ -597,6 +615,82 @@ public function save_tables()
...
+				}
+			}
+		}
+
+
+
+		$this->db->trans_complete();
+
+		$success = $this->db->trans_status();
+		$success = $result ? TRUE : FALSE;
```
Comment (daN4cat) [inline application/controllers/Config.php]: this doesn't look right to me, maybe it meant to be &= and not =

File: application/controllers/Config.php [hunk 2]
```diff
@@ -597,6 +615,82 @@ public function save_tables()
...
+			foreach($tax_categories as $tax_category)
+			{
+				if(!empty($tax_category['tax_category_id']) && !in_array($tax_category['tax_category_id'], $not_to_delete))
+				{
+					$this->Tax->delete_tax_category($tax_category['tax_category_id']);
+				}
+			}
+		}
+
+
```
Comment (daN4cat) [inline application/controllers/Config.php]: extra cr for no reason

File: application/views/configs/general_config.php [hunk 3]
```diff
@@ -345,16 +266,12 @@
...
-			{
-				number: "<?php echo $this->lang->line('config_default_tax_rate_number'); ?>"
-			},
-			default_sales_discount:
-			{
-				required: "<?php echo $this->lang->line('config_default_sales_discount_required'); ?>",
-				number: "<?php echo $this->lang->line('config_default_sales_discount_number'); ?>"
-			},
-			lines_per_page: 
+            default_sales_discount:
```
Comment (daN4cat) [inline application/views/configs/general_config.php]: indentation issue

File: application/views/configs/manage.php [hunk 4]
```diff
@@ -53,6 +56,9 @@
     <div class="tab-pane" id="general_tab">
         <?php $this->load->view("configs/general_config"); ?>
     </div>
+    <div class="tab-pane" id="tax_tab">
+		<?php $this->load->view("configs/tax_config"); ?>
```
Comment (daN4cat) [inline application/views/configs/manage.php]: extra tab

File: database/3.0.2_to_3.1.0.sql [hunk 5]
```diff
@@ -163,7 +163,7 @@ CREATE TABLE `ospos_sales_taxes` (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
 CREATE TABLE IF NOT EXISTS `ospos_tax_categories` (
-  `tax_category_id` int(10) NOT NULL,
+  `tax_category_id` int(10) NOT NULL AUTO_INCREMENT,
```
Comment (daN4cat) [inline database/3.0.2_to_3.1.0.sql]: it's better to add a statement with just this modification at the end of the file so who is using latest 3.1.0 can easily upgrade after this mergr

Comment (SteveIreland) [inline database/3.0.2_to_3.1.0.sql]: I didn't see your review notes until after I had already pushed it.  So I will make these changes and re-push.

**Comment by SteveIreland** [comment]
Okay, this should work.  However, it has a problem that I need some help with from more experienced eyes.  The updates seem to be working as expected.  I can't find any failures in the PHP code nor can I find any errors in the JavaScript.  However, I am no longer receiving the "Success" message.  When I debug it it clearly should be sending a success message.

Also, I modeled my changes after the Rewards package maintenance panel.  But when I had problems with my changes I found that the rewards package maintenance seems to have a problem too.  Attempts to delete a package appear to be successful at first but in reality are NOT successful.  

I would have corrected it, but since my changes are not yet 100% due to the missing Success message I didn't want to introduce my own problem into that code.

**Comment by daN4cat** [comment]
Strange I fixed the Rewards is sure with success message few days ago.  I'll check better but submit your changes and I'll fix both.

**Comment by SteveIreland** [comment]
I have two pull requests awaiting your acceptance.

**Comment by SteveIreland** [comment]
I should also mention that even though I changed the upgrade script to add the AUTO_INCREMENT as a subsequent step, the execution of that statement might fail unless you first clear the `tax_categories` table.

What I did was go into phpMyAdmin and exported the data from `tax_categories`, ran the script fragment, and then imported the data back in.

**Comment by daN4cat** [comment]
Thanks for all the work, I'll try to find the time next days to merge everything and leave master to last test and translation amendments.

**Comment by daN4cat** [comment]
I merged this, I tried to fix the issue related to saving of categories but it doesn't work.

I can see that the there is a common problem across all the similar story in rewards, tables and etc.

I'm not sure I can find the time but it needs fixing.

**Comment by odiea** [comment]
It appears that suspended quotes and suspended sales are now showing in Takings.

**Comment by SteveIreland** [comment]
@odiea Mine!  I'll take care of that  :-)

**Pull request merged**